### PR TITLE
Event Check-in - automatically check out time: minute wise (EXPOSUREAPP-6253)

### PR DESCRIPTION
**Pull request by kolyaopahle**
### Description
This PR changes the duration picker interval to 15 minute increments
it also rounds the default checkin time to the closest increment

### Steps to reproduce
1. Scan a qr code with none 15 minute based checkin duration
2. check if duration is rounded
3. check if picker is aligned to 15 minute increments

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/eventregistration/attendee/confirm/ConfirmCheckInViewModel.kt [hunk 1]
```diff
@@ -28,11 +29,14 @@ class ConfirmCheckInViewModel @AssistedInject constructor(
...
-        Duration.standardMinutes(
-            verifiedTraceLocation.traceLocation.defaultCheckInLengthInMinutes?.toLong() ?: 0L
-        )
+
+    private val truncatedDefaultCheckInLength = roundToNearestValidDuration(
+        verifiedTraceLocation.traceLocation.defaultCheckInLengthInMinutes ?: 0
     )
+    private val checkInLength = MutableStateFlow(truncatedDefaultCheckInLength)
+
+    private fun roundToNearestValidDuration(minutes: Int): Duration =
```
Comment (LukasLechnerDev) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/eventregistration/attendee/confirm/ConfirmCheckInViewModel.kt:38]: Some tests would be nice :)

**Review by LukasLechnerDev — APPROVED** [review]
Tested it and LGTM!

Some test for the rounding would be great :)

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/eventregistration/attendee/confirm/ConfirmCheckInViewModel.kt (hunk 1 above)
Comment (LukasLechnerDev) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/eventregistration/attendee/confirm/ConfirmCheckInViewModel.kt:38]: I will use some of your logic in one of my upcoming PRs, so I could also add some tests for the rounding if you like.

**Review by jurajkusnier — APPROVED** [review]
LGTM

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/eventregistration/attendee/confirm/ConfirmCheckInViewModel.kt [hunk 2]
```diff
@@ -28,11 +29,14 @@ class ConfirmCheckInViewModel @AssistedInject constructor(
...
+        Duration.standardMinutes((minutes.toFloat() / 15).roundToLong() * 15)
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/eventregistration/attendee/confirm/ConfirmCheckInViewModel.kt:39]: Could you please Extract `15` into a constant and give it a descriptive name

**Review by mtwalli — APPROVED** [review]
Small comment, no blocker for me

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/eventregistration/attendee/confirm/ConfirmCheckInViewModel.kt (hunk 2 above)
Comment (LukasLechnerDev) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/eventregistration/attendee/confirm/ConfirmCheckInViewModel.kt:39]: Will adapt this in my upcoming PR!

**Pull request merged**